gpt4all-bindings/python/gpt4all/pyllmodel.py

```diff
@@ -340,12 +341,18 @@ def run_llmodel_prompt(prompt: str, callback: ResponseCallbackType, **kwargs):
                 break
             yield response
 
+    def _callback_decoder(self, callback: RawResponseCallbackType) -> ResponseCallbackType:
+        def _callback(token_id: int, response: str) -> bool:
```

Comment:
Looks like there's a small mistake here:
```py
    def _callback_decoder(self, callback: ResponseCallbackType) -> RawResponseCallbackType:
        def _callback(token_id: int, response: bytes) -> bool:
            nonlocal callback
            return callback(token_id, response.decode("utf-8", "replace"))

        return _callback
```
The `_callback` is what gets passed into `ResponseCallback` and must have `bytes` as second argument. Consequently, it has the `RawResponseCallbackType`.

In my previous comment, I was thinking about renaming this one, first of all. Because it does the decoding.

Then the `callback` it calls gets the decoded bytes as second argument, i.e. a string.

Reply:
Ah, yes, you're right. Thank you for the `mypy` tool, it is actually really nice. 

The naming of this callback wrapper took me some time to think about. It is a functions that takes a callback with String and returns a callback with Bytes, so you can say that it "encodes" the callback. Or you can say that it's a decoder because it decodes the `response` argument. Depends on how you look at it.